python_modules/dagster/dagster/core/execution_plan/intermediates_manager.py

```diff
@@ -107,3 +108,59 @@ def set_value(self, step_output_handle, value):
 
     def has_value(self, step_output_handle):
         return self._files.has_file(*self._get_path_comps(step_output_handle))
+
+
+def check_path_comps(path_comps):
+    path_list = check.list_param(list(path_comps), 'path_comps', of_type=str)
+    check.param_invariant(path_list, 'path_list', 'Must have at least one comp')
+    return path_list
+
+
+class LocalTempFileStore:
+    def __init__(self, run_id):
+        check.str_param(run_id, 'run_id')
+        self.root = os.path.join('/tmp', 'dagster', 'runs', run_id, 'files')
+        self._created = False
+
+    def ensure_root_exists(self):
+        if not self._created:
+            mkdir_p(self.root)
+
+        self._created = True
```

Comment:
is this deliberate? if the root is deleted by some other process after `self._created` is set, we'll have errors

Reply:
yes this really doesn't help that much